File: TaskManager/TaskManager/Model/ToDo.cs
using System.ComponentModel.DataAnnotations;

namespace TaskManager.Model
{
    public class ToDo
    {
        public Guid Id { get; set; } = Guid.NewGuid();
        public string Title { get; set; }
        public string Description { get; set; }
        public DateTimeOffset Created { get; set; } = DateTimeOffset.UtcNow;
        public DateTimeOffset Updated { get; set; } = DateTimeOffset.UtcNow;
        public DateTimeOffset? DateLimit { get; set; } 
        public int Stats { get; set; }
        public bool IsDeleted { get; set; }
        public DateTimeOffset? DeleteAt { get; set; }
        public Guid IdUser { get; set; }
    }
}

File: TaskManagerFrontend/taskmanager-web/src/models/ToDo.ts
import type {ToDoStatusValue} from '../enum/Stats.ts'

export interface ToDo {
    id: string,
    title: string,
    description: string,
    created: string,
    updated: string,
    dateLimit: string,
    stats: ToDoStatusValue
}

export interface TodoSimple {
    id: string,
    title: string,
    created: string,
    dateLimit: string,
    stats: ToDoStatusValue
}

export interface CreateToDo {
    title: string,
    description: string,
    dateLimit: string | null,
    stats: number
}

export interface CreateToDoForm{
    title: string;
    description: string;
    dateLimit: string;
    stats: number;
}